# This is a BitKeeper generated diff -Nru style patch.
#
# ChangeSet
#   2005/01/29 14:25:32-02:00 james4765@cwazy.co.uk 
#   [PATCH] lcd: Add checks to CAP_SYS_ADMIN to potentially dangerous ioctl's
#   
#   This patch adds CAP_SYS_ADMIN checks to the potentially dangerous ioctls FLASH_Erase and FLASH_Burn
#   in the Cobalt LCD interface driver.
#   
#   Signed-off-by: James Nelson <james4765@gmail.com>
# 
# drivers/char/lcd.c
#   2005/01/27 21:54:01-02:00 james4765@cwazy.co.uk +4 -0
#   lcd: Add checks to CAP_SYS_ADMIN to potentially dangerous ioctl's
# 
diff -Nru a/drivers/char/lcd.c b/drivers/char/lcd.c
--- a/drivers/char/lcd.c	2005-02-03 05:58:28 -08:00
+++ b/drivers/char/lcd.c	2005-02-03 05:58:28 -08:00
@@ -386,6 +386,8 @@
 
 		int ctr=0;
 
+		if (!capable(CAP_SYS_ADMIN)) return -EPERM;
+
 		    // Chip Erase Sequence
 		WRITE_FLASH( kFlash_Addr1, kFlash_Data1 );
 		WRITE_FLASH( kFlash_Addr2, kFlash_Data2 );
@@ -421,6 +423,8 @@
 		
 
                 struct lcd_display display;
+
+		if (!capable(CAP_SYS_ADMIN)) return -EPERM;
 
                 if(copy_from_user(&display, (struct lcd_display*)arg, sizeof(struct lcd_display)))
 		  return -EFAULT;
